lib/src/dbus_ws_client.dart

```diff
@@ -0,0 +1,256 @@
+class DBusWSClient extends DBusClient {
+  WebSocketChannel? _channel;
+  final _readJsonBuffer = DBusWSReadBuffer();
+  final Uri _uri;
+  DBusWSClient(this._uri) : super(DBusAddress("unix:path=/bus")) {}
+
+
+  /// Terminates all active connections. If a client remains unclosed, the Dart process may not terminate.
+  @override
+  Future<void> close() async {
+    if (_channel != null) {
+      await _channel!.sink.close();
+      _channel = null;
+      _connectCompleter = null;
+    }
+  }
+
+  /// Open a socket connection to the D-Bus server.
+  @override
+  Future<void> _openSocket() async {
+    // print("> openSocket");
+    _channel = WebSocketChannel.connect(_uri);
+    _channel?.stream.listen(_processDataJson);
+  }
+
+  /// Performs authentication with D-Bus server.
+  @override
+  Future<bool> _authenticate() async {
+    return true;
+  }
+
+  /// Connects to the D-Bus server.
+  @override
+  Future<void> _connect() async {
+    // If already connecting, wait for that to complete.
+    if (_connectCompleter != null) {
+      return _connectCompleter?.future;
+    }
+    _connectCompleter = Completer();
+    await _openSocket();
+    _connectCompleter?.complete();
+  }
+
+  /// Adds a rule to match which messages to receive.
+  @override
+  Future<void> _addMatch(String rule) async {
+    var count = _matchRules[rule];
+    // print("> _addMatch $rule");
+    if (count == null) {
+      // print("> _addMatch callMethod: $rule");
+      _matchRules[rule] = 1;
+      await callMethod(
+          destination: 'org.freedesktop.DBus',
+          path: DBusObjectPath('/org/freedesktop/DBus'),
+          interface: 'org.freedesktop.DBus',
+          name: 'AddMatch',
+          values: [DBusString(rule)],
+          replySignature: DBusSignature(''));
+      // print("> _addMatch callMethod end");
+    } else {
+      _matchRules[rule] = count + 1;
+    }
+  }
+
+  /// Processes incoming data from the D-Bus server.
+  void _processDataJson(dynamic data) {
+    // print("_processDataJson: $data");
+    _readJsonBuffer.setResponseData(data);
+
+    var complete = false;
+    while (!complete) {
+      complete = _processMessages();
+      _readJsonBuffer.flush();
+    }
+  }
+
+  /// Processes messages (method calls/returns/errors/signals) received from the D-Bus server.
+  @override
+  bool _processMessages() {
+    var message = _readJsonBuffer.readMessage();
+    if (message == null) {
+      return true;
+    }
+
+    // print("> _processMessages: type: ${message.type}");
+    if (message.type == DBusMessageType.methodReturn ||
+        message.type == DBusMessageType.error) {
+      _processMethodResponse(message);
+    } else if (message.type == DBusMessageType.signal) {
+      _processSignal(message);
+    }
+
+    return true;
+  }
+
+  /// Processes a method return or error result from the D-Bus server.
+  @override
+  void _processMethodResponse(DBusMessage message) {
+    // Check has required fields.
```

Comment:
this method also seems identical to base _processMethodResponse

Reply:
There are some slight changes to base _processMethodResponse, so we need to keep the copy.